app/src/main/java/com/example/todo/HomePage.java

```diff
@@ -236,6 +236,7 @@ public static void listDeleted(int ID){
             if (dynamicList.get(i).getId() == ID) {
                 dynamicList.remove(i);
                 dynamicAdapter.notifyItemRemoved(i);
+                dynamicAdapter.notifyDataSetChanged();
```

Comment:
There are two inherent problems with notifyDataSetChanged. First is that this function call on an adapter refreshes the entire list, and hence this operation can get very expensive if your RecyclerView carries a large amount of data. Secondly, this function call does not perform or even allow for any animation. These two reasons make the use of notifyDataSetChanged a little less preferable.